code_challenges/code_wars/vowel_count.py

```diff
@@ -0,0 +1,7 @@
+# https://www.codewars.com/kata/54ff3102c1bad923760001f3
+
+
+def vowel_count(sentence):
+    vowels = ["a", "e", "i", "o", "u", "A", "E", "I", "O", "U"]
+    char_count = [char for char in sentence if char in vowels]
```

Comment:
Nice list comprehension practice!

If the input is sure to be small, this is fine.  Duplicating a long string into a list to get the count via it's length is less than ideal.

The boring version is a bit better for space complexity:
```python
count = 0
for letter in sentence:
   if letter in vowels:
       count += 1
```

In coding and a part of the interview process is knowing the time complexity and space complexity of your solution.   The vocabulary used to refer to this complexity is [big - O](https://stackabuse.com/big-o-notation-and-algorithm-analysis-with-python-examples/)

In you're PR's from now on make an educated guess about what the time and space complexity are in big-O notation.

Reply:
Hmmm...I actually read about this before when you sent me a link. And I think i get it. 

Would this be O(n+10) assuming vowels is the set 
`vowels = {"a", "e", "i", "o", "u", "A", "E", "I", "O", "U"}` 

Or does the count += 1 also get included since it's directly related to the size of sentence?

Reply:
The complexity is related to some aspect of what is passed in and really reflects the rate at which the memory/space or cpu cycles/time are increasing.  So maybe very technically this would be `O(n * 10)` this is reduced to `O(n)` because for every extra n from the input O amount of time is added to the function call worst case.

So for you solution the time complexity is `O(n)`, which is a nice goal to shoot for with solutions after you've already got one working regardless of complexity.  

The space complexity is also `O(n)` meaning that as the input increase in size the space needed to compute is increasing at the same rate (worst case).  Not making a new list reduces this to O(1) in space since keeping a count doesn't increase the memory usage past a single operation.